crates/bevy_ecs/src/system/query.rs

```diff
@@ -663,16 +695,16 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
     /// Rule of thumb: If the function body is (mostly) computationally expensive but there are not many items, a small batch size (=more batches) may help to even out the load.
     /// If the body is computationally cheap and you have many items, a large batch size (=fewer batches) avoids spawning additional futures that don't help to even out the load.
     ///
-    /// # Arguments
-    ///
-    ///* `batch_size` - The number of batches to spawn
-    ///* `f` - The function to run on each item in the query
+    /// [`ComputeTaskPool`]: bevy_tasks::prelude::ComputeTaskPool
     ///
     /// # Panics
-    /// The [`ComputeTaskPool`] is not initialized. If using this from a query that is being
-    /// initialized and run from the ECS scheduler, this should never panic.
     ///
-    /// [`ComputeTaskPool`]: bevy_tasks::prelude::ComputeTaskPool
+    /// This method panics if the [`ComputeTaskPool`] resource is added to the `World` before using this method.
+    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this should never panic.
```

Comment:
I think we should either guarantee this or leave it out if we can't

```suggestion
    /// If using this from a query that is being initialized and run from the [`Schedule`](crate::schedule::Schedule), this never panics.
```